src/core/Sampler/Sampler.cpp

```diff
@@ -233,6 +233,29 @@ void Sampler::noteOff(Note* pNote )
 }
 
 
+/// functions for pan parameters and laws-----------------
+float Sampler::getRatioPan( float fPan_L, float fPan_R ) {
+	// returns the single pan parameter in [-1,1] from the L,R gains
+	// It doesn't return ERROR if (L,R) = (0,0) nor if they are negative!!!!!
```

Comment:
Regarding the errors in those two functions: If there is no straight forward default value one can return/set, the C++ way of implementing errors would be an exception. But in Hydrogen it's almost never used (and introducing it would make the code most probably harder to maintain).

I would suggest to introduce these limits to `Instrument::set_pan` and `Note::set_pan` instead. If > 0, `0` will be assigned and a WARNINGLOG triggered and the same for 1 from above.

Reply:
`I would suggest to introduce these limits to Instrument::set_pan and Note::set_pan instead. If > 0, 0 will be assigned and a WARNINGLOG triggered and the same for 1 from above.`

Ok, I keep it for the final version

Reply:
> I would suggest to introduce these limits to Instrument::set_pan and Note::set_pan instead.

There is already a check for negative pan:
```
void Note::set_pan_l( float pan )
{
	__pan_l = check_boundary( pan, PAN_MIN, PAN_MAX );
}

```

In this way H2 one can set both pan_L = pan_R = 0. In this case there is a problem to reconvert back to the pan parameter. It could happen importing a bad song.